### PR TITLE
Correct ground temperature model index for Site:GroundDomain:Basement

### PR DESCRIPTION
**Pull request by mitchute**
A user reported the following:

_We've found that some configurations of Site:GroundDomain objects can crash v8.4. I've attached an IDF + EPW file which illustrates._

_The simulations run for slab domains and ground domains individually. Multiple domains aren’t a problem either, so long as they’re all one type of domain. The problem arises when you try to run a simulation with both slab and basement domains together._

Response:
The index for the ground temperature model was not being set correctly for the Site:GroundDomain:Basement. Likely a copy-paste error. It passed by because when running only basements, the total number of ground domains is equal to the number of basements. Adding different ground domain objects caused the numbering problem to show up and exposed it.

With this fix in place the user provided files run to completion.

### CODE REVIEW

**Comment by mjwitte** [comment]
Reference helpdesk ticket 10875

**Comment by Myoldmopar** [comment]
@mmatts This is a good fix, and super small.  What would it take to work up a unit test?  Could you pass in the input objects for a slab and a basement both and just call get input and check that the index is correct?

**Comment by mitchute** [comment]
@Myoldmopar Sure, I will work something up.

**Comment by mitchute** [comment]
@Myoldmopar could you review the unit test?

File: src/EnergyPlus/DataPlantPipingSystems.cc [hunk 1]
```diff
@@ -113,6 +113,18 @@ namespace DataPlantPipingSystems {
...
+	//*********************************************************************************************!
+
+	//*********************************************************************************************!
+
+	void
+	clear_state()
+	{
+		Array1D< FullDomainStructureInfo > PipingSystemDomains;
+		Array1D< PipeCircuitInfo > PipingSystemCircuits;
+		Array1D< PipeSegmentInfo > PipingSystemSegments;
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataPlantPipingSystems.cc]: Did you mean to call .deallocate on these?

Comment (Myoldmopar) [inline src/EnergyPlus/DataPlantPipingSystems.cc]: And just verify that there aren't any other lingering "module-level" variables to be cleaned up in preparation for subsequent unit tests.

**Comment by Myoldmopar** [comment]
I made a couple comments.  The unit test you put in should suffice fine.  Update based on the prior comments and we should be able to merge.

File: src/EnergyPlus/DataPlantPipingSystems.cc (hunk 1 above)
Comment (mitchute) [inline src/EnergyPlus/DataPlantPipingSystems.cc]: Yes, I did.

**Pull request merged**